packages/react-i18n/src/decorator.ts

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import {RegisterOptions} from './manager';
+import {useI18n} from './hooks';
+import I18n from './i18n';
+
+export interface WithI18nProps {
+  i18n: I18n;
+}
+
+export function withI18n(i18nOptions: RegisterOptions) {
+  return <P extends WithI18nProps>(
+    WrappedComponent: React.ComponentType<P>,
+  ): React.ComponentType<P> => props => {
+    const [i18n] = useI18n(i18nOptions);
```

Comment:
I think to truly replicate what existed before, this will need to also grab the `ShareTranslations` bit of the return value and render it around the wrapped component

Reply:
Added `< ShareTranslations></ShareTranslations>` wrapper.